x-pack/filebeat/module/zeek/notice/ingest/pipeline.json

```diff
@@ -0,0 +1,46 @@
+{
+  "description": "Pipeline for normalizing Zeek notice.log",
+  "processors": [
+    {
+      "script": {
+        "lang": "painless",
+        "source": "ctx.event.created = ctx['@timestamp']; ctx['@timestamp'] = (long)ctx['zeek']['notice']['ts'] * 1000; ctx.zeek.notice.remove('ts');"
+      }
+    },
+    {
+      "script": {
+        "lang": "painless",
+        "source": "ctx.destination.ip = ctx['destination']['address'];",
```

Comment:
Sorry for not catching this earlier, but this copy and the following two can be done with `set` and the same `if`.

We should avoid using Painless in cases like this, as it adds unnecessary overhead (even if small).

Check out an example in [Suricata module](https://github.com/elastic/beats/blob/7.0/x-pack/filebeat/module/suricata/eve/ingest/pipeline.json#L59-L65)

Reply:
Did not see there is an "if" for set - seems not in the documentations. And set processor was creating null value fields.